src/sentry/static/sentry/app/components/bases/pluginComponentBase.tsx

```diff
@@ -40,44 +51,46 @@ class PluginComponentBase extends React.Component {
     if (this.onSubmit) {
       this.onSubmit = this.onSave.bind(this, this.onSubmit.bind(this));
     }
-
     this.state = {
       state: FormState.READY,
-    };
-  }
-
-  UNSAFE_componentWillMount() {
-    this.api = new Client();
+    } as Readonly<S>;
   }
 
   componentWillUnmount() {
     this.api.clear();
   }
 
+  api = new Client();
+
+  fetchData() {
+    // Allow children to implement this
+  }
+
+  onSubmit() {
+    // Allow children to implement this
+  }
+
   onLoad(callback, ...args) {
     this.setState(
       {
         state: FormState.LOADING,
       },
-      callbackWithArgs(callback, ...args)
+      callbackWithArgs(this, callback, ...args)
     );
   }
 
-  onLoadSuccess(callback, ...args) {
-    this.setState(
-      {
-        state: FormState.READY,
-      },
-      callbackWithArgs(callback, ...args)
-    );
+  onLoadSuccess() {
```

Comment:
We don't actually pass a callback in and it was causing TS errors so I just removed that functionality